content/rust/concepts/collections/collections.md

```diff
@@ -0,0 +1,171 @@
+
+```rust
+    use std::collections::HashMap;
+
+    let mut scores = HashMap::new();
+
+    scores.insert(String::from("Blue"), 10);
+    scores.insert(String::from("Yellow"), 50);
+
+```
+
+## BTreeMap
+
+The `BTreeMap` is another data structure provided by the Rust standard library. It is a type of map that maintains its keys in sorted order. The name "BTreeMap" is derived from the underlying data structure it uses, which is a B-tree. The "B" in B-tree stands for "balanced," indicating that the tree structure is kept balanced to ensure efficient operations. It provides a key-value mapping similar to HashMap but with the added benefit of keys being sorted. The data type can be created in the following way:
+
+```rust
+let mut btree_map: BTreeMap<&str, i32> = BTreeMap::new(); // Create an empty BTreeMap
+
+btree_map.insert("one", 1);     //Insert key-value pairs
+btree_map.insert("three", 3);   //Insert key-value pairs
+
+println!("Value for key 'two': {:?}", btree_map.get("two")); // Access and print values
+```
+
+## Set
+
+In Rust, the standard library provides a `HashSet` collection, which is an implementation of a set. A set is a collection of unique elements, and the HashSet ensures that each element appears only once.`BTreeSet` is another collection provided by the standard library. It is a set implemented using a B-tree, just like BTreeMap is a map implemented using a B-tree. The `BTreeSet` maintains a sorted set of unique elements. Here's an example of creating and using a `HashSet` and `BTreeSet`:
+
+```rust
+use std::collections::{HashSet, BTreeSet};
+
+fn main() {
+    // HashSet example
+    let mut hash_set: HashSet<i32> = [3, 1, 2].iter().cloned().collect();
+    hash_set.remove(&2);
+    println!("Is 2 in HashSet? {}", hash_set.contains(&2));
+    println!("HashSet elements: {:?}", hash_set);
+
+    // BTreeSet example
+    let mut btree_set: BTreeSet<i32> = [3, 1, 2].iter().cloned().collect();
+    btree_set.remove(&2);
+    println!("Is 2 in BTreeSet? {}", btree_set.contains(&2));
+    println!("BTreeSet elements: {:?}", btree_set);
+}
+```
+
+## BinaryHeap
+
+A `binaryHeap` is a data structure provided by the standard library in the std::collections module. It is useful for scenarios where you need quick access to the maximum element. Elements are ordered such that the parent is always greater than or equal to its children. The data type can be created in the following way:
+
+```rust
+use std::collections::BinaryHeap;
+
+fn main() {
+    // Create a max binary heap
+    let mut max_heap = BinaryHeap::from(vec![4, 2, 8, 5]);
+
+    // Insert elements into the heap
+    max_heap.push(10);
+    max_heap.push(1);
+
+    // Pop and print elements (max to min)
+    while let Some(max_element) = max_heap.pop() {
+        println!("{}", max_element);
+    }
+}
+```
```

Comment:
The mentioned collections can have their entries so can you please remove them? You can add more relevant content to the concept.

Reply:
Hello @avdhoottt just checking. You mean i should delete this collections from the entry  BinaryHeap, Set ,BTreeMap ,HashMap and String ?